data/blog/2025/2025-03-18-mezzio101-what-defines-a-middleware-architecture.md

```diff
@@ -0,0 +1,123 @@
+---
+id: 2025-03-18-mezzio101-what-defines-a-middleware-architecture
+author: bidi
+title: 'Mezzio101: What Defines a Middleware Architecture?'
+draft: false
+public: true
+created: '2025-03-18T11:00:00-01:00'
+updated: '2025-03-18T11:00:00-01:00'
+tags:
+  - middleware
+  - software architecture
+  - framework
+openGraphImage: '2025-03-18-mezzio101-what-defines-a-middleware-architecture.png'
+openGraphDescription: 'Mezzio101: What Defines a Middleware Architecture?'
+---
+
+Nowadays, applications are getting more and more complex.
+The classic way of executing sequential code, from **request to response**, still works for simple apps, but using a modern architecture has many benefits.
+Enter the **middleware**, an intermediary code that processes the request in various ways before it even reaches your custom code.
+
+<!--- EXTENDED -->
+
+In this article we will list the advantages of using middlewares in your applications.
```

Comment:
```suggestion
In this article we will list the advantages of using middleware in your applications.
```

The plural of "middleware" is "middleware"; please review and correct throughout.